### PR TITLE
Handle ObjectGUID claims

### PR DESCRIPTION
**Pull request by blizzz**
They are base64-encoded binary string. 

Also drops Nc14 support, due to failing integration tests with unclear cause. It is EOL anyhow. Version bump to 2.5 included.

### CODE REVIEW

**Comment by blizzz** [comment]
meanwhile i think the failing tests for 14 were hiccups.. Anyway, let the dead rest in peace, shall we?

Does it look OK for you?

**Comment by blizzz** [comment]
Got positive feedback from a customer  👍

File: lib/UserBackend.php [hunk 1]
```diff
@@ -691,4 +691,46 @@ public function updateAttributes($uid,
...
+		if(preg_match('/^[a-f0-9]{8}(-[a-f0-9]{4}){4}[a-f0-9]{8}$/i', $candidate) === 1) {
+			$uid = $candidate;
+		}
+		return $uid;
+	}
+
+	/**
+	 * @see \OCA\User_LDAP\Access::convertObjectGUID2Str
+	 */
+	public function convertObjectGUID2Str($oguid) {
```
Comment (skjnldsv) [inline lib/UserBackend.php]: Does it needs to be public?

**Pull request merged**